The job **`aarch64-gnu-llvm-20-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/17837990546/job/50719973282) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/50719973282) [(plain)](https://github.com/rust-lang/rust/commit/8f9724232571d5406cd541ccf0996aab514b1fae/checks/50719973282/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
------FileCheck stderr------------------------------
/checkout/tests/codegen-llvm/vec_pop_push_noop.rs:9:16: error: CHECK-NOT: excluded string found in input
 // CHECK-NOT: call
               ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/vec_pop_push_noop/vec_pop_push_noop.ll:22:7: note: found here
 tail call void @llvm.assume(i1 %cond.i)
      ^~~~
/checkout/tests/codegen-llvm/vec_pop_push_noop.rs:11:16: error: CHECK-NOT: excluded string found in input
 // CHECK-NOT: call
               ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/vec_pop_push_noop/vec_pop_push_noop.ll:22:7: note: found here
 tail call void @llvm.assume(i1 %cond.i)
      ^~~~

Input file: /checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/vec_pop_push_noop/vec_pop_push_noop.ll
Check file: /checkout/tests/codegen-llvm/vec_pop_push_noop.rs

-dump-input=help explains the following input dump.

Input was:
<<<<<<
        1: ; ModuleID = 'vec_pop_push_noop.2c8a275f269cfdaf-cgu.0' 
        2: source_filename = "vec_pop_push_noop.2c8a275f269cfdaf-cgu.0" 
        3: target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i8:8:32-i16:16:32-i64:64-i128:128-n32:64-S128-Fn32" 
        4: target triple = "aarch64-unknown-linux-gnu" 
        5:  
        6: @alloc_3f54897a36e7c663fd135cd0aec64be4 = private unnamed_addr constant [50 x i8] c"/checkout/tests/codegen-llvm/vec_pop_push_noop.rs\00", align 1 
        7: @alloc_8ea69435bd230cbb5d6874e6936f9f54 = private unnamed_addr constant <{ ptr, [16 x i8] }> <{ ptr @alloc_3f54897a36e7c663fd135cd0aec64be4, [16 x i8] c"1\00\00\00\00\00\00\00\16\00\00\00\07\00\00\00" }>, align 8 
        8:  
        9: ; Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(argmem: read, inaccessiblemem: write) uwtable 
       10: define void @noop(ptr noalias nocapture noundef readonly align 8 dereferenceable(24) %v) unnamed_addr #0 personality ptr @rust_eh_personality { 
       11: start: 
       12:  %0 = getelementptr inbounds nuw i8, ptr %v, i64 16 
       13:  %_2.i = load i64, ptr %0, align 8, !alias.scope !2, !noundef !5 
       14:  %.not = icmp eq i64 %_2.i, 0 
       15:  br i1 %.not, label %bb3, label %"_ZN5alloc3vec16Vec$LT$T$C$A$GT$8push_mut17h592ac3131ad73c3bE.exit" 
       16:  
       17: "_ZN5alloc3vec16Vec$LT$T$C$A$GT$8push_mut17h592ac3131ad73c3bE.exit": ; preds = %start 
       18:  %_3.0.i = add i64 %_2.i, -1 
       19:  %1 = getelementptr inbounds nuw i8, ptr %v, i64 8 
       20:  %self1.i = load i64, ptr %1, align 8, !range !6, !alias.scope !2, !noundef !5 
       21:  %cond.i = icmp ult i64 %_3.0.i, %self1.i 
       22:  tail call void @llvm.assume(i1 %cond.i) 
not:9            !~~~                                error: no match expected
not:11           !~~~                                error: no match expected
       23:  br label %bb3 
       24:  
       25: bb3: ; preds = %start, %"_ZN5alloc3vec16Vec$LT$T$C$A$GT$8push_mut17h592ac3131ad73c3bE.exit" 
       26:  ret void 
       27: } 
       28:  
       29: ; Function Attrs: uwtable 
       30: define void @push_byte(ptr noalias noundef align 8 dereferenceable(24) %v) unnamed_addr #1 personality ptr @rust_eh_personality { 
       31: start: 
       32:  %0 = getelementptr inbounds nuw i8, ptr %v, i64 16 
       33:  %len.i = load i64, ptr %0, align 8, !alias.scope !7, !noalias !10, !noundef !5 
       34:  %1 = getelementptr inbounds nuw i8, ptr %v, i64 8 
       35:  %self1.i = load i64, ptr %1, align 8, !range !6, !alias.scope !7, !noalias !10, !noundef !5 
       36:  %_4.i = icmp eq i64 %len.i, %self1.i 
       37:  br i1 %_4.i, label %bb1.i, label %"_ZN5alloc3vec16Vec$LT$T$C$A$GT$8push_mut17h592ac3131ad73c3bE.exit" 
       38:  
       39: bb1.i: ; preds = %start 
       40: ; call alloc::raw_vec::RawVec<T,A>::grow_one 
       41:  tail call void @"_ZN5alloc7raw_vec19RawVec$LT$T$C$A$GT$8grow_one17hc2398457852dcea9E"(ptr noalias noundef nonnull align 8 dereferenceable(24) %v, ptr noalias noundef nonnull readonly align 8 dereferenceable(24) @alloc_8ea69435bd230cbb5d6874e6936f9f54) 
       42:  br label %"_ZN5alloc3vec16Vec$LT$T$C$A$GT$8push_mut17h592ac3131ad73c3bE.exit" 
       43:  
       44: "_ZN5alloc3vec16Vec$LT$T$C$A$GT$8push_mut17h592ac3131ad73c3bE.exit": ; preds = %start, %bb1.i 
       45:  %_18.i = load ptr, ptr %v, align 8, !alias.scope !7, !noalias !10, !nonnull !5, !noundef !5 
       46:  %2 = getelementptr inbounds nuw i8, ptr %_18.i, i64 %len.i 
       47:  store i8 3, ptr %2, align 1 
       48:  %_13.0.i = add i64 %len.i, 1 
       49:  store i64 %_13.0.i, ptr %0, align 8, !alias.scope !7, !noalias !10 
       50:  ret void 
       51: } 
       52:  
       53: ; Function Attrs: nounwind uwtable 
       54: declare noundef range(i32 0, 10) i32 @rust_eh_personality(i32 noundef, i32 noundef, i64 noundef, ptr noundef, ptr noundef) unnamed_addr #2 
       55:  
       56: ; Function Attrs: mustprogress nocallback nofree nosync nounwind willreturn memory(inaccessiblemem: write) 
       57: declare void @llvm.assume(i1 noundef) #3 
       58:  
       59: ; alloc::raw_vec::RawVec<T,A>::grow_one 
       60: ; Function Attrs: noinline uwtable 
       61: declare void @"_ZN5alloc7raw_vec19RawVec$LT$T$C$A$GT$8grow_one17hc2398457852dcea9E"(ptr noalias noundef align 8 dereferenceable(16), ptr noalias noundef readonly align 8 dereferenceable(24)) unnamed_addr #4 
       62:  
       63: attributes #0 = { mustprogress nofree norecurse nosync nounwind willreturn memory(argmem: read, inaccessiblemem: write) uwtable "frame-pointer"="non-leaf" "probe-stack"="inline-asm" "target-cpu"="generic" "target-features"="+v8a,+outline-atomics" } 
       64: attributes #1 = { uwtable "frame-pointer"="non-leaf" "probe-stack"="inline-asm" "target-cpu"="generic" "target-features"="+v8a,+outline-atomics" } 
       65: attributes #2 = { nounwind uwtable "frame-pointer"="non-leaf" "probe-stack"="inline-asm" "target-cpu"="generic" "target-features"="+v8a,+outline-atomics" } 
       66: attributes #3 = { mustprogress nocallback nofree nosync nounwind willreturn memory(inaccessiblemem: write) } 
       67: attributes #4 = { noinline uwtable "frame-pointer"="non-leaf" "probe-stack"="inline-asm" "target-cpu"="generic" "target-features"="+v8a,+outline-atomics" } 
       68:  
       69: !llvm.module.flags = !{!0} 
       70: !llvm.ident = !{!1} 
       71:  
       72: !0 = !{i32 8, !"PIC Level", i32 2} 
       73: !1 = !{!"rustc version 1.92.0-nightly (0ede12ba3 2025-09-18)"} 
       74: !2 = !{!3} 
       75: !3 = distinct !{!3, !4, !"_ZN5alloc3vec16Vec$LT$T$C$A$GT$3pop17h537df8ee561e53a3E: %self"} 
       76: !4 = distinct !{!4, !"_ZN5alloc3vec16Vec$LT$T$C$A$GT$3pop17h537df8ee561e53a3E"} 
       77: !5 = !{} 
       78: !6 = !{i64 0, i64 -9223372036854775808} 
       79: !7 = !{!8} 
       80: !8 = distinct !{!8, !9, !"_ZN5alloc3vec16Vec$LT$T$C$A$GT$8push_mut17h592ac3131ad73c3bE: %self"} 
       81: !9 = distinct !{!9, !"_ZN5alloc3vec16Vec$LT$T$C$A$GT$8push_mut17h592ac3131ad73c3bE"} 
       82: !10 = !{!11} 
       83: !11 = distinct !{!11, !9, !"_ZN5alloc3vec16Vec$LT$T$C$A$GT$8push_mut17h592ac3131ad73c3bE: argument 1"} 
>>>>>>

------------------------------------------

error: verification with 'FileCheck' failed
status: exit status: 1
command: "/usr/lib/llvm-20/bin/FileCheck" "--input-file" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/vec_pop_push_noop/vec_pop_push_noop.ll" "/checkout/tests/codegen-llvm/vec_pop_push_noop.rs" "--check-prefix=CHECK" "--allow-unused-prefixes" "--dump-input-context" "100"
stdout: none
--- stderr -------------------------------
/checkout/tests/codegen-llvm/vec_pop_push_noop.rs:9:16: error: CHECK-NOT: excluded string found in input
 // CHECK-NOT: call
               ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/vec_pop_push_noop/vec_pop_push_noop.ll:22:7: note: found here
 tail call void @llvm.assume(i1 %cond.i)
      ^~~~
/checkout/tests/codegen-llvm/vec_pop_push_noop.rs:11:16: error: CHECK-NOT: excluded string found in input
 // CHECK-NOT: call
               ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/vec_pop_push_noop/vec_pop_push_noop.ll:22:7: note: found here
 tail call void @llvm.assume(i1 %cond.i)
      ^~~~

Input file: /checkout/obj/build/aarch64-unknown-linux-gnu/test/codegen-llvm/vec_pop_push_noop/vec_pop_push_noop.ll
Check file: /checkout/tests/codegen-llvm/vec_pop_push_noop.rs

-dump-input=help explains the following input dump.

Input was:
<<<<<<
        1: ; ModuleID = 'vec_pop_push_noop.2c8a275f269cfdaf-cgu.0' 
        2: source_filename = "vec_pop_push_noop.2c8a275f269cfdaf-cgu.0" 
        3: target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i8:8:32-i16:16:32-i64:64-i128:128-n32:64-S128-Fn32" 
        4: target triple = "aarch64-unknown-linux-gnu" 
        5:  
        6: @alloc_3f54897a36e7c663fd135cd0aec64be4 = private unnamed_addr constant [50 x i8] c"/checkout/tests/codegen-llvm/vec_pop_push_noop.rs\00", align 1 
        7: @alloc_8ea69435bd230cbb5d6874e6936f9f54 = private unnamed_addr constant <{ ptr, [16 x i8] }> <{ ptr @alloc_3f54897a36e7c663fd135cd0aec64be4, [16 x i8] c"1\00\00\00\00\00\00\00\16\00\00\00\07\00\00\00" }>, align 8 
        8:  
        9: ; Function Attrs: mustprogress nofree norecurse nosync nounwind willreturn memory(argmem: read, inaccessiblemem: write) uwtable 
       10: define void @noop(ptr noalias nocapture noundef readonly align 8 dereferenceable(24) %v) unnamed_addr #0 personality ptr @rust_eh_personality { 
       11: start: 
       12:  %0 = getelementptr inbounds nuw i8, ptr %v, i64 16 
       13:  %_2.i = load i64, ptr %0, align 8, !alias.scope !2, !noundef !5 
       14:  %.not = icmp eq i64 %_2.i, 0 
       15:  br i1 %.not, label %bb3, label %"_ZN5alloc3vec16Vec$LT$T$C$A$GT$8push_mut17h592ac3131ad73c3bE.exit" 
       16:  
       17: "_ZN5alloc3vec16Vec$LT$T$C$A$GT$8push_mut17h592ac3131ad73c3bE.exit": ; preds = %start 
       18:  %_3.0.i = add i64 %_2.i, -1 
       19:  %1 = getelementptr inbounds nuw i8, ptr %v, i64 8 
       20:  %self1.i = load i64, ptr %1, align 8, !range !6, !alias.scope !2, !noundef !5 
       21:  %cond.i = icmp ult i64 %_3.0.i, %self1.i 
       22:  tail call void @llvm.assume(i1 %cond.i) 
not:9            !~~~                                error: no match expected
not:11           !~~~                                error: no match expected
       23:  br label %bb3 
       24:  
       25: bb3: ; preds = %start, %"_ZN5alloc3vec16Vec$LT$T$C$A$GT$8push_mut17h592ac3131ad73c3bE.exit" 
       26:  ret void 
       27: } 
       28:  
       29: ; Function Attrs: uwtable 
       30: define void @push_byte(ptr noalias noundef align 8 dereferenceable(24) %v) unnamed_addr #1 personality ptr @rust_eh_personality { 
       31: start: 
       32:  %0 = getelementptr inbounds nuw i8, ptr %v, i64 16 
       33:  %len.i = load i64, ptr %0, align 8, !alias.scope !7, !noalias !10, !noundef !5 
       34:  %1 = getelementptr inbounds nuw i8, ptr %v, i64 8 
       35:  %self1.i = load i64, ptr %1, align 8, !range !6, !alias.scope !7, !noalias !10, !noundef !5 
       36:  %_4.i = icmp eq i64 %len.i, %self1.i 
       37:  br i1 %_4.i, label %bb1.i, label %"_ZN5alloc3vec16Vec$LT$T$C$A$GT$8push_mut17h592ac3131ad73c3bE.exit" 
       38:  
       39: bb1.i: ; preds = %start 
       40: ; call alloc::raw_vec::RawVec<T,A>::grow_one 
       41:  tail call void @"_ZN5alloc7raw_vec19RawVec$LT$T$C$A$GT$8grow_one17hc2398457852dcea9E"(ptr noalias noundef nonnull align 8 dereferenceable(24) %v, ptr noalias noundef nonnull readonly align 8 dereferenceable(24) @alloc_8ea69435bd230cbb5d6874e6936f9f54) 
       42:  br label %"_ZN5alloc3vec16Vec$LT$T$C$A$GT$8push_mut17h592ac3131ad73c3bE.exit" 
       43:  
       44: "_ZN5alloc3vec16Vec$LT$T$C$A$GT$8push_mut17h592ac3131ad73c3bE.exit": ; preds = %start, %bb1.i 
       45:  %_18.i = load ptr, ptr %v, align 8, !alias.scope !7, !noalias !10, !nonnull !5, !noundef !5 
       46:  %2 = getelementptr inbounds nuw i8, ptr %_18.i, i64 %len.i 
       47:  store i8 3, ptr %2, align 1 
       48:  %_13.0.i = add i64 %len.i, 1 
       49:  store i64 %_13.0.i, ptr %0, align 8, !alias.scope !7, !noalias !10 
       50:  ret void 
       51: } 
       52:  
       53: ; Function Attrs: nounwind uwtable 
       54: declare noundef range(i32 0, 10) i32 @rust_eh_personality(i32 noundef, i32 noundef, i64 noundef, ptr noundef, ptr noundef) unnamed_addr #2 
       55:  
       56: ; Function Attrs: mustprogress nocallback nofree nosync nounwind willreturn memory(inaccessiblemem: write) 
       57: declare void @llvm.assume(i1 noundef) #3 
       58:  
       59: ; alloc::raw_vec::RawVec<T,A>::grow_one 
       60: ; Function Attrs: noinline uwtable 
       61: declare void @"_ZN5alloc7raw_vec19RawVec$LT$T$C$A$GT$8grow_one17hc2398457852dcea9E"(ptr noalias noundef align 8 dereferenceable(16), ptr noalias noundef readonly align 8 dereferenceable(24)) unnamed_addr #4 
       62:  
       63: attributes #0 = { mustprogress nofree norecurse nosync nounwind willreturn memory(argmem: read, inaccessiblemem: write) uwtable "frame-pointer"="non-leaf" "probe-stack"="inline-asm" "target-cpu"="generic" "target-features"="+v8a,+outline-atomics" } 
       64: attributes #1 = { uwtable "frame-pointer"="non-leaf" "probe-stack"="inline-asm" "target-cpu"="generic" "target-features"="+v8a,+outline-atomics" } 
       65: attributes #2 = { nounwind uwtable "frame-pointer"="non-leaf" "probe-stack"="inline-asm" "target-cpu"="generic" "target-features"="+v8a,+outline-atomics" } 
       66: attributes #3 = { mustprogress nocallback nofree nosync nounwind willreturn memory(inaccessiblemem: write) } 
       67: attributes #4 = { noinline uwtable "frame-pointer"="non-leaf" "probe-stack"="inline-asm" "target-cpu"="generic" "target-features"="+v8a,+outline-atomics" } 
       68:  
       69: !llvm.module.flags = !{!0} 
       70: !llvm.ident = !{!1} 
       71:  
       72: !0 = !{i32 8, !"PIC Level", i32 2} 
       73: !1 = !{!"rustc version 1.92.0-nightly (0ede12ba3 2025-09-18)"} 
       74: !2 = !{!3} 
       75: !3 = distinct !{!3, !4, !"_ZN5alloc3vec16Vec$LT$T$C$A$GT$3pop17h537df8ee561e53a3E: %self"} 
       76: !4 = distinct !{!4, !"_ZN5alloc3vec16Vec$LT$T$C$A$GT$3pop17h537df8ee561e53a3E"} 
       77: !5 = !{} 
       78: !6 = !{i64 0, i64 -9223372036854775808} 
       79: !7 = !{!8} 
       80: !8 = distinct !{!8, !9, !"_ZN5alloc3vec16Vec$LT$T$C$A$GT$8push_mut17h592ac3131ad73c3bE: %self"} 
       81: !9 = distinct !{!9, !"_ZN5alloc3vec16Vec$LT$T$C$A$GT$8push_mut17h592ac3131ad73c3bE"} 
       82: !10 = !{!11} 
       83: !11 = distinct !{!11, !9, !"_ZN5alloc3vec16Vec$LT$T$C$A$GT$8push_mut17h592ac3131ad73c3bE: argument 1"} 
>>>>>>
------------------------------------------

---- [codegen] tests/codegen-llvm/vec_pop_push_noop.rs stdout end ----

```

</details>

--- a/tests/codegen-llvm/issues/issue-122600-ptr-discriminant-update.rs
+++ b/tests/codegen-llvm/issues/issue-122600-ptr-discriminant-update.rs
@@ -1,4 +1,7 @@
 //@ compile-flags: -Copt-level=3
+//@ revisions: new old
+//@ [old] ignore-llvm-version: 22 - 99
+//@ [new] min-llvm-version: 22
 
 #![crate_type = "lib"]
 
@@ -22,8 +25,8 @@ pub unsafe fn update(s: *mut State) {
     // CHECK-NOT: memcpy
     // CHECK-NOT: 75{{3|4}}
 
-    // CHECK: %[[TAG:.+]] = load i8, ptr %s, align 1
-    // CHECK-NEXT: trunc nuw i8 %[[TAG]] to i1
+    // old: %[[TAG:.+]] = load i8, ptr %s, align 1
+    // old-NEXT: trunc nuw i8 %[[TAG]] to i1
 
     // CHECK-NOT: load
     // CHECK-NOT: store

--- a/tests/codegen-llvm/vec_pop_push_noop.rs
+++ b/tests/codegen-llvm/vec_pop_push_noop.rs
@@ -1,4 +1,7 @@
 //@ compile-flags: -Copt-level=3
+//@ revisions: new old
+//@ [old] ignore-llvm-version: 22 - 99
+//@ [new] min-llvm-version: 22
 
 #![crate_type = "lib"]
 
@@ -7,7 +10,7 @@
 pub fn noop(v: &mut Vec<u8>) {
     // CHECK-NOT: grow_one
     // CHECK-NOT: call
-    // CHECK: tail call void @llvm.assume
+    // old: tail call void @llvm.assume
     // CHECK-NOT: grow_one
     // CHECK-NOT: call
     // CHECK: {{ret|[}]}}

--- a/tests/codegen-llvm/vecdeque_pop_push.rs
+++ b/tests/codegen-llvm/vecdeque_pop_push.rs
@@ -1,4 +1,7 @@
 //@ compile-flags: -Copt-level=3
+//@ revisions: new old
+//@ [old] ignore-llvm-version: 22 - 99
+//@ [new] min-llvm-version: 22
 
 #![crate_type = "lib"]
 
@@ -8,7 +11,7 @@ use std::collections::VecDeque;
 // CHECK-LABEL: @noop_back(
 pub fn noop_back(v: &mut VecDeque<u8>) {
     // CHECK-NOT: grow
-    // CHECK: tail call void @llvm.assume
+    // old: tail call void @llvm.assume
     // CHECK-NOT: grow
     // CHECK: ret
     if let Some(x) = v.pop_back() {

